pkg/daemon/daemon.go

```diff
@@ -498,20 +498,23 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	}
 
 	if reqDrain {
-		if !dn.disableDrain {
-			ctx, cancel := context.WithCancel(context.TODO())
-			defer cancel()
-
-			glog.Infof("nodeStateSyncHandler(): get drain lock for sriov daemon")
-			done := make(chan bool)
-			go dn.getDrainLock(ctx, done)
-			<-done
-		}
+		if !dn.isNodeDraining() {
```

Comment:
This prevents daemon to go into `go dn.getDrainLock`, what if the other node is already in the go routine while the first node releases leader election lock?

I'm thinking the following scenario:
t1: node-1 gets leader election lock and node-2 is waiting for leader election to be released (node-2 daemon is already in the go routine loop for getting drain lock)
t2: node-1 daemon release leader election lock and reboot
t3: node-2 doesn't check line 501 since it is already in the go routine and gets the leader election lock.

Reply:
you're right, I didn't notice that `isNodeDraining` checks for Draining annotations to avoid re-entering this clause after rebooting.